subworkflows/local/utils_nfcore_rnaseq_pipeline/main.nf

```diff
@@ -545,29 +527,79 @@ def biotypeInGtf(gtf_file, biotype) {
 }
 
 //
-// Function that parses and returns the predicted strandedness from the RSeQC infer_experiment.py output
+// Function to determine library type by comparing type counts. Consistent
+// between Salmon and RSeQC
 //
-def getInferexperimentStrandedness(inferexperiment_file, cutoff=30) {
-    def sense        = 0
-    def antisense    = 0
-    def undetermined = 0
+
+def calculateStrandedness(forwardFragments, reverseFragments, undeterminedFragments, threshold) {
+    def totalFragments = forwardFragments + reverseFragments + undeterminedFragments
+    def totalStrandedFragments = forwardFragments + reverseFragments
+
+    def strandedness = 'undetermined'
+    if (totalStrandedFragments > 0) {
+        def forwardProportion = forwardFragments / (totalStrandedFragments as double)
+        def reverseProportion = reverseFragments / (totalStrandedFragments as double)
+        def proportionDifference = Math.abs(forwardProportion - reverseProportion)
+
+        if (forwardProportion >= threshold) {
+            strandedness = 'forward'
+        } else if (reverseProportion >= threshold) {
+            strandedness = 'reverse'
+        } else if (proportionDifference <= (1 - threshold)) {
+            strandedness = 'unstranded'
+        }
+    }
+
+    return [
+        inferred_strandedness: strandedness,
+        forwardFragments: forwardFragments / (totalFragments as double),
+        reverseFragments: reverseFragments / (totalFragments as double),
+        undeterminedFragments: undeterminedFragments / (totalFragments as double)
+    ]
+}
+
+//
+// Function that parses Salmon quant 'lib_format_counts.json' output file to get inferred strandedness
+//
+
+def getSalmonInferredStrandedness(json_file, threshold = 0.9) {
```

Comment:
Default threshold here is `0.9`, is this the same threshold as `strand_predict_threshold` params ?
If so, why `0.9`?
why not `0.8`?